app/src/main/cpp/ControllerContainer.cpp

```diff
@@ -203,38 +204,42 @@ void ControllerContainer::SetHandJointLocations(const int32_t aControllerIndex,
 
-        CreationContextPtr create = m.context.lock();
-
-        // Initialize left hand action button, which for now triggers back navigation.
-        // Note that Quest's runtime already shows the hamburger menu button when left
-        // hand is facing head.
+    // Initialize left hand action button, which for now triggers back navigation.
+    // Note that Quest's runtime already shows the hamburger menu button when left
+    // hand is facing head.
 #if !defined(OCULUSVR)
-        if (controller.leftHanded) {
-            TextureGLPtr texture = create->LoadTexture("menu.png");
-            assert(texture);
-            texture->SetTextureParameter(GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-            texture->SetTextureParameter(GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-
-            const float iconWidth = 0.03f;
-            const float aspect = (float)texture->GetWidth() / (float)texture->GetHeight();
-
-            QuadPtr icon = Quad::Create(create, iconWidth, iconWidth / aspect, nullptr);
-            icon->SetTexture(texture, texture->GetWidth(), texture->GetHeight());
-            icon->UpdateProgram("");
-
-            controller.handActionButtonToggle = Toggle::Create(create);
-            controller.handActionButtonTransform = Transform::Create(create);
-            controller.handActionButtonToggle->AddNode(controller.handActionButtonTransform);
-            controller.handActionButtonTransform->AddNode(icon->GetRoot());
-            controller.handActionButtonToggle->ToggleAll(false);
-            m.root->AddNode(controller.handActionButtonToggle);
-        }
+    if (controller.leftHanded && controller.handActionButtonToggle == nullptr) {
+        TextureGLPtr texture = create->LoadTexture("menu.png");
+        assert(texture);
+        texture->SetTextureParameter(GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        texture->SetTextureParameter(GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+
+        const float iconWidth = 0.03f;
+        const float aspect = (float)texture->GetWidth() / (float)texture->GetHeight();
+
+        QuadPtr icon = Quad::Create(create, iconWidth, iconWidth / aspect, nullptr);
+        icon->SetTexture(texture, texture->GetWidth(), texture->GetHeight());
+        icon->UpdateProgram("");
+
+        controller.handActionButtonToggle = Toggle::Create(create);
+        controller.handActionButtonTransform = Transform::Create(create);
+        controller.handActionButtonToggle->AddNode(controller.handActionButtonTransform);
+        controller.handActionButtonTransform->AddNode(icon->GetRoot());
+        controller.handActionButtonToggle->ToggleAll(false);
+        m.root->AddNode(controller.handActionButtonToggle);
+    }
 #endif
 
+    // Due to unstable or inaccurate hand-tracking data, hand mesh models
+    // are not enabled on Pico devices. We are drawing a sphere for each joint
+    // instead.
+#if defined(PICOXR)
+    // Initialize hand joints if needed
+    if (controller.handJointTransforms.size() == 0) {
```

Comment:
Hmm why is this needed now? Looks like it might be a different bug?

Reply:
We had that before too. It is the way we use to know that we have not initialized the hand joint transforms yet. Only that now it is only done for PicoXR.